lib/src/network/backends.rs

```diff
@@ -188,21 +201,133 @@ impl BackendList {
       })
   }
 
-  pub fn available_backends(&mut self) -> Vec<&mut Rc<RefCell<Backend>>> {
-    self.backends.iter_mut()
-      .filter(|backend| (*backend.borrow()).can_open())
-      .collect()
-  }
-
   pub fn next_available_backend(&mut self) -> Option<&mut Rc<RefCell<Backend>>> {
-    let mut backends:Vec<&mut Rc<RefCell<Backend>>> = self.available_backends();
-    if backends.is_empty() {
-      return None;
+    if self.backends.len() == 0 {
+      return None
     }
 
-    let rnd = random::<usize>();
-    let idx = rnd % backends.len();
+    self.load_balancing.next_available_backend(&mut self.backends)
```

Comment:
if the list of backends is not filtered before passing to the load balancing algorithm (as done in https://github.com/sozu-proxy/sozu/pull/459/files#diff-0231d86414182a60450d3bce3504028bL193 ) then circuit breaking won't work anymore